### DISCUSSION TITLE
Can janusgraph support the ClickHouse?

### DISCUSSION THREAD

**Discussion by ggservice007** [JanusGraph/janusgraph#1370] [Ideas]
I love janusgraph and ClickHouse.

https://github.com/yandex/ClickHouse/issues/4001

We can together make it. 

So nice.

**Comment by porunov** [comment]
@ggservice007 as far as I know, right now JanusGraph doesn't support `ClickHouse` as a storage backend. I believe it is possible to add support for ClickHouse but for this we need someone who is an expert in both JanusGraph and ClickHouse and is willing to contribute to JanusGraph. If you wish to contribute to JanusGraph by adding a new supported backend to it then JanusGraph community can help you with it. 
You can ask development questions in the JanusGraph developers group: https://groups.google.com/forum/#!forum/janusgraph-dev

**Comment by ggservice007** [comment]
@porunov 
So nice.
I will try it.

**Comment by blinkov** [comment]
Briefly browsed around source/docs and it sounds like JanusGraph uses those backends just for read/write queries for specific keys. If that's right there's not much sense to add ClickHouse to that backends list, because it's optimized for batch writes and analytical retrieval queries (with aggregation, joins, sorts, etc.) which is completely different workload.

**Comment by ggservice007** [comment]
@porunov 

Thanks  a lot.

Maybe we can write a new store format for the ClickHouse.

**Comment by ggservice007** [comment]
@porunov  
Any idea about that "Briefly browsed around source/docs and it sounds like JanusGraph uses those backends just for read/write queries for specific keys. "

**Comment by porunov** [comment]
@ggservice007 I am not familiar with ClickHouse and don't know if it is wise to use it as a storage backend. All I know is that it is a column-oriented database management system. If so, then it is possible to use it as a storage backend. But to understand if you **should** use it as a storage backend you need to dig into ClickHouse.

**Comment by blinkov** [comment]
@porunov what kind of queries are typically issued to those backends?

**Comment by porunov** [comment]
@blinkov I didn't research yet how JanusGraph proceed queries to the storage backend. I know what kind of queries go into index backends (cursor is used to retrieve search data in batches. Search queries are not key specific). I assume that there are not only queries for specific keys when working with storage backends.
You can ask this question JanusGraph community here:
https://groups.google.com/forum/#!forum/janusgraph-users
Possibly someone can give you an answer
